probe-rs/src/debug/variable_cache.rs

```diff
@@ -42,11 +42,16 @@ impl Serialize for VariableCache {
         fn recurse_cache(variable_cache: &VariableCache) -> VariableTreeNode {
             let root_node = variable_cache.root_variable();
 
+            let children_count = variable_cache
+                .get_children(root_node.variable_key)
+                .unwrap()
+                .len();
+
             VariableTreeNode {
                 name: root_node.name.clone(),
                 type_name: root_node.type_name.clone(),
                 value: root_node.get_value(variable_cache),
-                children: if root_node.range_upper_bound > 50 {
+                children: if children_count > 50 {
```

Comment:
I know this is pre-existing but either the comment is wrong, or we could do better, or both.